lib/Core/BuildEngine.cpp

```diff
@@ -528,7 +535,10 @@ class BuildEngineImpl {
   void processRuleScanRequest(RuleScanRequest request) {
     auto& ruleInfo = *request.ruleInfo;
 
-    assert(ruleInfo.isScanning());
+    // With forced builds in cycle breaking, we may end up being asked to scan
+    // something that has already been 'scanned'
+    if (!ruleInfo.isScanning())
+      return;
```

Comment:
Is there a way we could preserve this assert so it checks that either the rule is scanning, or it is being forced?

Reply:
I'm not sure I understand when this happens, can you explain?

Reply:
I believe the pathway that leads here is that a node gets placed on the deferredScanRequests list. When we force a node to build it has explicitly been moved past the scanning phase.  However, the node that deferred the scan may not be in the cycle we are currently evaluating.